### PR TITLE
Permitted warnings about root-dir for all versions of Cylc.

### PR DESCRIPTION
**Pull request by wxtim**
Pull request closing an issue documented here:

### Issue

This issue (https://github.com/cylc/cylc-rose/issues/230) describes a rose suite configuration which raises warnings when cylc compatibility mode is turned off. The warning that `root-dir` is no longer valid at Cylc 8 is valid whether or not compatibility mode is turned on and should be shown in either case.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Comment by wxtim** [comment]
This is also broken on 8.1.x.

**Review by oliver-sanders — APPROVED** [review]
Tested with https://github.com/cylc/cylc-flow/pull/5582, Rose warnings show as expected.

**Comment by oliver-sanders** [comment]
Note this contains a commit under review on https://github.com/cylc/cylc-rose/pull/232, please merge after.

File: tests/conftest.py [hunk 1]
```diff
@@ -100,9 +100,7 @@ def pytest_runtest_makereport(item, call):
 def _cylc_validate_cli(capsys, caplog):
     """Access the validate CLI"""
     def _inner(srcpath, args=None):
-        parser = validate_gop()
-        options = Options(parser, args)()
-
+        options = Options(validate_gop(), args)()
```
Comment (oliver-sanders) [inline tests/conftest.py:103]: MB:

```suggestion
        parser = validate_gop()
        options = Options(parser, args)()
```

File: tests/conftest.py [hunk 2]
```diff
@@ -100,12 +100,7 @@ def pytest_runtest_makereport(item, call):
 def _cylc_validate_cli(capsys, caplog):
     """Access the validate CLI"""
     def _inner(srcpath, args=None):
-        parser = Options(validate_gop())
-        options = parser()
-
-        if args is not None:
-            options.__dict__.update(args)
-
+        options = Options(validate_gop(), args)()
```
Comment (oliver-sanders) [inline tests/conftest.py:103]: MB:

```suggestion
        parser = validate_gop()
        options = Options(parser, args)()
````

**Comment by oliver-sanders** [comment]
@wxtim, I made a mistake deconflicting this, with the diff in the comment above this should work.

**Review by MetRonnie — APPROVED** [review]
I've sorted out the merge issue. Awaiting tests passing

**Pull request merged**